src/System.Text.Json/tests/JsonDocumentTests.cs

```diff
@@ -3736,16 +3735,13 @@ public static void Disposing_On_Multiple_Threads_Should_Not_Return_To_ArrayPool_
 
             Task.WaitAll(tasks);
 
-            // When ArrayPool gets corrupted, the Rent method might pull an already rented array, which is incorrect.
-            // So we will pull as many arrays as calls to JsonElement.Dispose and check they are unique.
-            HashSet<IntPtr> uniqueAddresses = new HashSet<IntPtr>();
+            // When ArrayPool gets corrupted, the Rent method might return an already rented array, which is incorrect.
+            // So we will rent as many arrays as calls to JsonElement.Dispose and check they are unique.
+            HashSet<byte[]> uniqueAddresses = new HashSet<byte[]>();
```

Comment:
nit: you could use var here, which avoids having to update the T in multiple places (within HashSet).

Reply:
Alternatively, you could continue to not use `var` and make me happier :smile: